corehq/apps/users/views/__init__.py

```diff
@@ -372,6 +375,7 @@ def post(self, request, *args, **kwargs):
             return self.get(request, *args, **kwargs)
 
 
+@location_safe
```

Comment:
This is safe because the locations widget is already safe, right?  However, that's client-side only, so I think this needs a server-side validation that:

* Confirms that the locations assigned are accessible (in case a malicious actor futzed with the request manually)
* At least one (accessible) location has been assigned

Or does that already come for free from the mobile worker side of things?

Reply:
Huh, I think you're right.  It looks like it is done in the new user form in `clean_location_id`, but I don't see anything in the `clean_` methods of `CommtrackUserForm`.

Reply:
If a location-restricted user doesn't have a location assigned, then they'll get a [403 error](https://github.com/dimagi/commcare-hq/blob/ae87eb2ebdd1a06437438327ebd02dfa913d6f83/corehq/apps/locations/middleware.py#L37-L38) when they access I think any page.  Though I guess maybe we don't have to prevent that, if we don't do currently when unrestricted web users edit restricted users.  I'm not sure, still seems good to keep them in their section of the tree, otherwise they'll immediately lose access to editing that web user.  Might be a question to loop in product on.

Reply:
Eh, I guess the same question also occurs for restricted users editing mobile workers, so we should probably just mirror that user-facing behavior.  I do think separate server-side validation of that behavior is best practice though